Fixes Operator Reply Request Indicator Bit Length

olabusayoT opened the pull request:
- this is a 1 bit field that was marked as having 2 bits
- added unparser tests for C and D1
- bumped version to 0.0.7

#16

olabusayoT committed:
Fixes Operator Reply Request Indicator Bit Length

- this is a 1 bit field that was marked as having 2 bits
- added unparser tests for C and D1
- bumped version to 0.0.7

#16

mbeckerle commented on src/test/resources/com/tresys/mil-std-2045/milstd2045.tdml:
```diff
@@ -539,6 +719,204 @@ SOFTWARE.
+    </tdml:infoset>
+
+    <tdml:document bitOrder="LSBFirst">
+      <!-- From K02.1_1.1.VMB modified to have zero message size. -->
```
Is this comment left over from a cut/paste? 

Update to mention this is modified and has operator_reply_request_indicator with value 1.

mbeckerle commented on src/test/resources/com/tresys/mil-std-2045/milstd2045.tdml:
```diff
@@ -335,6 +335,186 @@ SOFTWARE.
     </tdml:infoset>
   </tdml:parserTestCase>
 
+  <tdml:unparserTestCase name="test2045msghdr1_unparsed"
```
Add comment to document? Or make name more descriptive of what this test is specifically to exercise?

mbeckerle commented on src/test/resources/com/tresys/mil-std-2045/milstd2045.tdml:
```diff
@@ -335,6 +335,186 @@ SOFTWARE.
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+
+    <tdml:document bitOrder="LSBFirst">
```
If this data is specifically to show that operator_reply_request_indicator is properly implemented, then you need to specify which byte contains that bit. 

A good way to do this would be to use a documentPart type="bits" for that one byte of the data. The rest can stay as the hex you have.

mbeckerle commented on src/test/resources/com/tresys/mil-std-2045/milstd2045.tdml:
```diff
@@ -335,6 +335,186 @@ SOFTWARE.
+    </tdml:document>
+  </tdml:unparserTestCase>
+
+  <tdml:unparserTestCase name="test2045msghdr1_variant_unparsed"
```
Variant to illustrate what? Add doc comment or improve name.

mbeckerle commented on src/test/resources/com/tresys/mil-std-2045/milstd2045.tdml:
```diff
@@ -335,6 +335,186 @@ SOFTWARE.
+      </tdml:dfdlInfoset>
+    </tdml:infoset>
+
+    <tdml:document bitOrder="LSBFirst">
```
As above, if this is to illustrate hving operator_reply_request_indicator right, then make that one byte of this data be a documentPart type="bits" so you can break out exactly which bit is that one.

mbeckerle commented on src/test/resources/com/tresys/mil-std-2045/milstd2045.tdml:
```diff
@@ -539,6 +719,204 @@ SOFTWARE.
     </tdml:infoset>
   </tdml:parserTestCase>
 
+  <tdml:unparserTestCase name="test_2045_C_msghdr1_unparsed" root="milstd2045_application_header"
```
As above. Doc comment, or improve name.

mbeckerle commented on src/test/resources/com/tresys/mil-std-2045/milstd2045.tdml:
```diff
@@ -539,6 +719,204 @@ SOFTWARE.
+    </tdml:infoset>
+
+    <tdml:document bitOrder="LSBFirst">
+      <!-- From K02.1_1.1.VMB modified to have zero message size. -->
```
comment still relevant? (If so keep it, but possibly add to it to explain the additional data mod for this test.

mbeckerle commented on src/test/resources/com/tresys/mil-std-2045/milstd2045.tdml:
```diff
@@ -539,6 +719,204 @@ SOFTWARE.
+    </tdml:document>
+  </tdml:unparserTestCase>
+
+  <tdml:unparserTestCase name="test_2045_C_msghdr1_variant_unparsed" root="milstd2045_application_header"
```
variant for what?

mbeckerle reviewed (approved):
+1 I have some suggestions for doc comment additions, or you can just make the test names more explanatory, though, with field names like operator_reply_request_indicator, it's pretty hard to make a short test name.  Also would be good to break out specific bits tests are intended to illustrate.

olabusayoT committed:
Fixes Operator Reply Request Indicator Bit Length

- updated to split into bits relevant for operator_reply_request and machine_acknowledge_request indicators
- updated test names and added comments to make test purpose clearer

#16

mbeckerle reviewed (approved):
+1
